src/parser/jobs/ast/modules/ArcanaTracking/ArcanaSuggestions.tsx

```diff
@@ -56,172 +57,126 @@ interface SleeveIcon {
+	private tryCloseWindow(event: Events['statusRemove']) {
+		const lastWindow: DivinationWindow | undefined = _.last(this.history)
 
-		// only track the things one player added
-		if (event.sourceID && event.sourceID !== this.parser.player.id) { return }
-
-		const lastWindow: DIVWindow | undefined = _.last(this.history)
-
-		if (!lastWindow) {
+		if (lastWindow == null) {
 			return
 		}
 
 		// Cache whether we've seen a buff removal event for this status, just in case they happen at exactly the same timestamp
-		lastWindow.buffsRemoved.push(event.ability.guid)
+		lastWindow.buffsRemoved.push(event.status)
 
 		if (this.isWindowOkToClose(lastWindow)) {
 			lastWindow.end = event.timestamp
 		}
 	}
 
 	// Make sure all applicable statuses have fallen off before the window closes
-	private isWindowOkToClose(window: DIVWindow): boolean {
+	private isWindowOkToClose(window: DivinationWindow): boolean {
 		if (!window.buffsRemoved.includes(this.data.statuses.DIVINATION.id)) {
 			return false
 		}
 		return true
 	}
-
-	private onCast(event: CastEvent) {
-		const lastWindow: DIVWindow | undefined = _.last(this.history)
-
-		// If we don't have a window, bail
-		if (!lastWindow) {
-			return
-		}
-
-		const action = this.data.getAction(event.ability.guid)
-
-		// Can't do anything else if we didn't get a valid action object
-		if (!action) {
-			return
-		}
-
-		// If this window isn't done yet add the action to the list
-		if (!lastWindow.end) {
-			lastWindow.rotation.push(event)
-
-			if (action.onGcd) {
-				lastWindow.gcdCount++
-			}
-			if (lastWindow.playersBuffed < 1) {
-				lastWindow.containsOtherAST = true
-			}
-			return
-		}
-
-		// If we haven't recorded a trailing GCD event for this closed window, do so now
-		if (lastWindow.end && !lastWindow.trailingGcdEvent && action.onGcd) {
-			lastWindow.trailingGcdEvent = event
-		}
-	}
 	//end divination functions
 
 	private _onComplete() {
-		const combatants = this.combatants.getEntities()
-		for (const [, combatant] of Object.entries(combatants)) {
-			if (combatant.type === 'LimitBreak') {
-				continue
-			}
-			this.partyComp.push(combatant.type)
-		}
-
 		this.cardLogs = this.arcanaTracking.cardLogs.map(artifact => {
-			const targetId = artifact.lastEvent.type !== 'init'
-				? artifact.lastEvent.targetID
-				: undefined
-			const target = targetId !== this.parser.player.id
-				? this.combatants.getEntity(targetId)
-				: this.combatants.selected
-
+			if (artifact.lastEvent.type === 'action') {
+				const targetId = artifact.lastEvent.target
+				const target = this.actors.get(targetId)
+				const cardLog: CardLog = {
+					...artifact,
+					targetName: target.name,
+					targetJob: target.job,
+				}
+				return cardLog
+			}
 			const cardLog: CardLog = {
 				...artifact,
-				targetName: target ? target.name : null,
-				targetJob: target ? target.type : null,
+				targetName: '',
+				targetJob: 'UNKNOWN',
 			}
 			return cardLog
```

Comment:
I tested doing that, but then target and targetId also have to be adjusted and then a bunch of artifact.lastEvent.type === 'action' has to be put and it looks cluttery. if I store artifact.lastEvent.type === 'action' as a variable, ts erroneously thinks lastEvent could be death even if I use a ternary before it

TL;DR: left this because I think it looks nicer than the alternative because everything else

Reply:
Fair enough!